lib/public_include/fty_shm.h

```diff
@@ -77,9 +88,21 @@ class shmMetrics
     // If you use this, DO NOT DELETE the fty_proto_t. It will be take
     // care by the shmlMetrics's destructor.
     //  (same warning if you access to it using iterator)
+
+    // @param index : index of the m_metricsVector
+    // @return : the address of the indexed element
     fty_proto_t*      get(int index);
+
+    // duplicates the index element
+    // @param index : index of the m_metricsVector
+    // @return : duplicated element
     fty_proto_t*      getDup(int index);
+
+    // pushes back the input fty_proto_t pointer
+    // @param metric : member will be pushed back to the vector
     void              add(fty_proto_t* metric);
+
+    // returns size of the vector
     long unsigned int size();
```

Comment:
fanciful "long unsigned int"
shall be size_t... but don't change that as it is a lib api...
same for get()  getDup() that take "int" argq... must be size_t to...